src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
+        method: "GET",
+        headers: { accept: "application/json" }
+      });
+      const response = await fetchUrl;
+      const myData = await response.json();
+      this.setState({
+        jokeData: { results: myData.results },
+        loadingState: "loaded"
+      });
+    } catch (e) {
+      this.setState({ loadingState: "Something went wrong" });
     }
   }
 
-  async componentDidMount() {
-    const url = 'https://icanhazdadjoke.com/search?term="cat"';
-    const fetchUrl = fetch(url, {
-      method: "GET",
-      headers: { accept: "application/json" }
-    });
-    const response = await fetchUrl;
-    const myData = await response.json();
-    this.setState({ jokeData: { results: myData.results } });
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.loadingState === "loading") {
+      this.fetchJokes();
+    }
+  }
+
+  componentDidMount() {
+    this.fetchJokes();
   }
 
   // This needs rendering inside a map function, then rendering in the render as <div>{jokeList}</div>
   // <Joke key="[ENTER THE KEY HERE]" joke={this.state}/>
   render() {
     const jokeList = this.state.jokeData.results.map(item => (
       <Joke key={item.id} joke={item.joke} />
-    ))
+    ));
     return (
       <div className="App">
-        <PageHeader />
+        <PageHeader>
+          <SearchInput
+            handleChange={this.handleChange}
+            handleClick={this.handleClick}
+            placeholder="Search for a joke..."
+            onChange={this.handleChange}
+            value={this.state.searchTerm}
+          />
+        </PageHeader>
         <div className="joke-wrapper">
-          {jokeList}
+          {this.state.loadingState === "loaded"
+            ? jokeList
+            : `${this.state.loadingState}`}
         </div>
       </div>
     );
   }
 }
 
+SearchInput.propTypes = {
```

Comment:
If you're going to use `propTypes` you should set them up where `SearchInput` is declared ("SearchInput.jsx " in this case). Doing it this way is confusing and potentially dangerous. You could be overriding the existing `propTypes` set up in "SearchInput.jsx "

Reply:
They're gone for now